specs/rating/rating.md

```diff
@@ -0,0 +1,162 @@
+        <slot
+            ${slotted({
+                property: "slottedRadioButtons",
+                filter: elements("[role=radio]"),
+            })}
+        ></slot>
+    </div>
+    <slot name="label"></slot>
+</template>
+```
+
+## Implementation
+
+```html
+<fast-rating>
+    <fast-rating-item value="1"></fast-rating-item>
+    <fast-rating-item value="2"></fast-rating-item>
+    <fast-rating-item value="3"></fast-rating-item>
+    <fast-rating-item value="4"></fast-rating-item>
+    <fast-rating-item value="5"></fast-rating-item>
+</fast-rating>
+```
+
+### Accessibility
+
+Keyboard interaction
+
+- Tab and Shift + Tab will move focus in and out of the rating component. When focus moves onto the rating component, the first item is higlighted and selected.
+- Right and Down Arrow will move focus to the next rating item in the group. As focused is moved to the next item, both the focused item and the previous item are visually hilighted, but the the focused item is only selected.
+- Left and Up Arrow will move focus to the previous radio button in the group, removing the visual highlight from the previously focused item.
+
+
+Assistive Technology
+
+- When focused is on the component, the reader will announce the component, the current item and the the total amount of items. Example: "rating, 1 of 5 stars".
+As focused is moved to the next item, the reader will read out the current value and the amount of total items. Example: "2 of 5 stars, 3 of 5 stars, 4 of 5 stars, 5 of 5 stars".
+
+
+
+# Rating item
+
+## Overview
+The `fast-rating-item` is to used inside the `fast-rating` component.
+
+### API
+
+Extends [Radio](../radio/radio.spec.md).
+
+_Component name:_
+
+- `fast-rating-item`
+
+_Attributes:_
+
+- `disabled` - A boolean to remove interaction on the item.
+- `readonly` - A boolean to remove interaction on the item.
+
+_Slots:_
+
+- `unchecked-icon` - The element that is visible when the rating item is unchecked.
+- `checked-icon` - The element that is visible when the rating item is checked.
+
+### Anatomy and Appearance
+
+The Rating item template is extended from Radio.
+
+_Template:_
+
+```html
+<template
+        role="radio"
+        class="${x => (x.checked ? "checked" : "")} ${x =>
+            x.readOnly ? "readonly" : ""}"
+        aria-checked="${x => x.checked}"
+        aria-required="${x => x.required}"
+        aria-disabled="${x => x.disabled}"
+        aria-readonly="${x => x.readOnly}"
+        @keypress="${(x, c) => x.keypressHandler(c.event as KeyboardEvent)}"
+        @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+    >
+        <div class="rating-icons" part="rating-icons">
+            <slot name="unchecked-icon" part="unchecked-icon" aria-hidden="true">
+                ${definition.uncheckedIcon || ""}
+            </slot>
+            <slot name="checked-icon" part="checked-icon" aria-hidden="true">
+                ${definition.checkedIcon || ""}
+            </slot>
+        </div>
```

Comment:
Can you elaborate a bit about how checked / unchecked maps to rating - is there a method of addressing partial ratings with this approach?

Reply:
I ran through a couple of names for these slots. I had front-icon/back-icon. empty-icon/filled-icon. I am not tied to unchecked-icon/checked-icon, but the reasoning for these names is because these items were basically radio items, so I went with a name that corresponds with the input being checked or unchecked.